src/backend/renderer/pixman/mod.rs

```diff
@@ -0,0 +1,1183 @@
+//! Implementation of the rendering traits using pixman
+use std::{
+    collections::HashSet,
+    rc::Rc,
+    sync::atomic::{AtomicBool, Ordering},
+};
+
+use drm_fourcc::{DrmFormat, DrmFourcc, DrmModifier};
+use pixman::{Filter, FormatCode, Image, Operation, Repeat};
+use tracing::warn;
+
+use crate::{
+    backend::allocator::{
+        dmabuf::{DmaBufSyncFlags, Dmabuf, DmabufMapping, WeakDmabuf},
+        format::has_alpha,
+        Buffer,
+    },
+    utils::{Buffer as BufferCoords, Physical, Rectangle, Scale, Size, Transform},
+};
+
+#[cfg(feature = "wayland_frontend")]
+use crate::{
+    backend::renderer::{ImportDmaWl, ImportMemWl},
+    wayland::{compositor::SurfaceData, shm},
+};
+#[cfg(feature = "wayland_frontend")]
+use wayland_server::{protocol::wl_buffer, Resource, Weak};
+
+#[cfg(all(
+    feature = "wayland_frontend",
+    feature = "backend_egl",
+    feature = "use_system_lib"
+))]
+use super::ImportEgl;
+use super::{
+    sync::SyncPoint, Bind, DebugFlags, ExportMem, Frame, ImportDma, ImportMem, Offscreen, Renderer, Texture,
+    TextureFilter, TextureMapping, Unbind,
+};
+
+mod error;
+
+pub use error::*;
+
+lazy_static::lazy_static! {
```

Comment:
I think if we're willing to uprev the rust dependency to 1.70.0, OnceCell should contain all the functionality you need.

Reply:
created a separate issue for this